src/functions/avl-validate/index.ts

```diff
@@ -0,0 +1,157 @@
+import { createServerErrorResponse, createValidationErrorResponse } from "@bods-integrated-data/shared/api";
+import { createAuthorizationHeader } from "@bods-integrated-data/shared/avl/utils";
+import { getDate } from "@bods-integrated-data/shared/dates";
+import { logger } from "@bods-integrated-data/shared/logger";
+import {
+    avlServiceDeliverySchema,
+    avlServiceRequestSchema,
+    avlValidateRequestSchema,
+} from "@bods-integrated-data/shared/schema/avl-validate.schema";
+import { InvalidXmlError } from "@bods-integrated-data/shared/validation";
+import { APIGatewayEvent, APIGatewayProxyResult } from "aws-lambda";
+import axios, { AxiosError } from "axios";
+import { XMLBuilder, XMLParser } from "fast-xml-parser";
+import { ZodError, z } from "zod";
+
+const requestBodySchema = z
+    .string({
+        required_error: "Body is required",
+        invalid_type_error: "Body must be a string",
+    })
+    .transform((body) => JSON.parse(body))
+    .pipe(avlValidateRequestSchema);
+
+const generateServiceRequestMessage = (currentTimestamp: string) => {
+    const serviceRequestJson = {
+        ServiceRequest: {
+            RequestTimestamp: currentTimestamp,
+            RequestorRef: "BODS",
+            VehicleMonitoringRequest: {
+                VehicleMonitoringRequest: {
+                    RequestTimestamp: currentTimestamp,
+                },
+            },
+        },
+    };
+
+    const verifiedServiceRequest = avlServiceRequestSchema.parse(serviceRequestJson);
+
+    const completeObject = {
+        "?xml": {
+            "#text": "",
+            "@_version": "1.0",
+            "@_encoding": "UTF-8",
+            "@_standalone": "yes",
+        },
+        Siri: {
+            "@_version": "2.0",
+            "@_xmlns": "http://www.siri.org.uk/siri",
+            "@_xmlns:ns2": "http://www.ifopt.org.uk/acsb",
+            "@_xmlns:ns3": "http://www.ifopt.org.uk/ifopt",
+            "@_xmlns:ns4": "http://datex2.eu/schema/2_0RC1/2_0",
+            ServiceRequest: {
```

Comment:
it would be more readable to directly add `ServiceRequest` here, then add the nested version attribute afterwards as `completeObject.Siri.ServiceRequest.VehicleMonitoringRequest.VehicleMonitoringRequest["@_version"] = "2.0"`